tests/osaka/eip7692_eof_v1/eip4750_functions/test_callf_execution.py

```diff
@@ -22,6 +22,85 @@
 pytestmark = pytest.mark.valid_from(EOF_FORK_NAME)
 
 
+def test_callf_factorial(eof_state_test: EOFStateTestFiller):
+    """Test factorial implementation with recursive CALLF instructions"""
+    eof_state_test(
+        data=Container(
+            sections=[
+                Section.Code(
+                    Op.CALLDATALOAD(0) + Op.SSTORE(0, Op.CALLF[1]) + Op.STOP,
+                    max_stack_height=2,
+                ),
+                Section.Code(
+                    Op.PUSH1[1]
+                    + Op.DUP2
+                    + Op.GT
+                    + Op.RJUMPI[4]
+                    + Op.POP
+                    + Op.PUSH1[1]
+                    + Op.RETF
+                    + Op.PUSH1[1]
+                    + Op.DUP2
+                    + Op.SUB
+                    + Op.CALLF[1]
+                    + Op.DUP2
+                    + Op.MUL
+                    + Op.SWAP1
+                    + Op.POP
+                    + Op.RETF,
+                    code_inputs=1,
+                    code_outputs=1,
+                    max_stack_height=3,
+                ),
+            ]
+        ),
+        tx_data=(57).to_bytes(32, "big"),
```

Comment:
Should we parameterize a few more values?  A smaller ones like 1 (to show recursive base case) and 5 (an easy one to calculate by hand) and also 58 to show what happens with overflow?

Reply:
Parametrized. But the are number of options to go here (e.g. you can skip result param and compute it inside the test). Feel free to modify it if you have other suggestions.